chia/full_node/full_node.py

```diff
@@ -765,9 +761,16 @@ async def new_peak(self, request: full_node_protocol.NewPeak, peer: WSChiaConnec
                 if await self.short_sync_batch(peer, uint32(max(curr_peak_height - 6, 0)), request.height):
                     return None
 
+            if self._sync_bool:
+                return None
```

Comment:
iiuc, this is the key change, really. I expect you could just check to see if `self._sync_task is not None` here. except, I suppose we would need to check if the task is complete, and join it and clear the task handle somewhere as well.

I think kyle's comment seems relevant. It seems risky to perform the sync inside the call. Current and/or future callers of `new_peak()` may not expect it to be held up for the full sync duration.